### PR TITLE
Migrate git/sha.go to the new world

### PR DESCRIPTION
**Pull request by kevgo**
This requires injecting variables and propagating errors in a few more places.

### CODE REVIEW

File: src/drivers/gitlab.go [hunk 1]
```diff
@@ -8,12 +8,14 @@ import (
 	"github.com/git-town/git-town/src/drivers/helpers"
 )
 
+// gitlabConfig defines the configuration data needed by the GitLab driver.
 type gitlabConfig interface {
```
Comment (charlierudolph) [inline src/drivers/gitlab.go:12]: Should this be replaced by the new shared `config`?

File: src/steps/run.go [hunk 2]
```diff
@@ -63,7 +63,11 @@ func Run(runState *RunState, repo *git.ProdRepo, driver drivers.CodeHostingDrive
 				exitWithMessages(runState.UnfinishedDetails.CanSkip)
 			}
 		}
-		runState.UndoStepList.Prepend(step.CreateUndoStep())
+		undoSteps, err := step.CreateUndoStep(repo)
```
Comment (charlierudolph) [inline src/steps/run.go:66]: `undoSteps` should be renamed to `undoStep` as it is singular

File: src/drivers/core.go [hunk 3]
```diff
@@ -27,6 +25,21 @@ type CodeHostingDriver interface {
 	HostingServiceName() string
 }
 
+// config defines the configuration data needed by the driver package.
+type config interface {
```
Comment (kevgo) [inline src/drivers/core.go:29]: In order to remove the dependency on the `git` package, I have unified all driver-specific interfaces into this interface.

File: src/drivers/gitlab.go (hunk 1 above)
Comment (kevgo) [reply src/drivers/gitlab.go]: Yes, that was an omission.

File: src/steps/run.go (hunk 2 above)
Comment (kevgo) [reply src/steps/run.go]: done

**Pull request merged**